File: TechTest/app/src/main/java/com/david_kapas/android/techtest/presentation/posts/router/PostListRouter.java
package com.david_kapas.android.techtest.presentation.posts.router;

import android.arch.lifecycle.LifecycleOwner;

/**
 * PostList router interface for navigation from ListActivity.
 * Created by David Kapas on 2018.03.17.
 */

public interface PostListRouter extends LifecycleOwner {

    void showGeneralErrorDialog();

    void openPost(int postId);
}

File: TechTest/app/src/main/java/com/david_kapas/android/techtest/presentation/posts/router/PostListRouter.kt
package com.david_kapas.android.techtest.presentation.posts.router

import android.arch.lifecycle.LifecycleOwner

/**
 * PostList router interface for navigation from ListActivity.
 * Created by David Kapas on 2018.03.17.
 */

interface PostListRouter : LifecycleOwner {

    fun showGeneralErrorDialog()

    fun openPost(postId: Int)
}